File: iosApp/iosApp/Presentation/Main/CafeManagement/CafeInfoEdit/CafeInfoEditAction.swift
//
//  CafeInfoEditAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation

enum CafeInfoEditAction {
    case clickBack
    case changeCafeName(String)
    case changeCafeDescription(String)
    case changeConceptType(String)
    case changeAddress(String)
    case changeContactNumber(String)
    case changeWeekdayOpen(String)
    case changeWeekdayClose(String)
    case changeWeekendOpen(String)
    case changeWeekendClose(String)
    case clickRepresentativeImage
    case selectRepresentativeImage(String)
    case clickAddGalleryImage
    case addGalleryImage(String)
    case removeGalleryImage(Int)
    case clickPinLocation
    case setPinnedLocation(latitude: Double, longitude: Double)
    case clickManageExceptionDates
    case dismissImageRequiredAlert
    case clickSave
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/cafeinfo/CafeInfoEditAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement.cafeinfo

sealed interface CafeInfoEditAction {
    data object ClickBack : CafeInfoEditAction
    data class ChangeCafeName(val value: String) : CafeInfoEditAction
    data class ChangeCafeDescription(val value: String) : CafeInfoEditAction
    data class ChangeAddress(val value: String) : CafeInfoEditAction
    data class ChangeContactNumber(val value: String) : CafeInfoEditAction
    data class ChangeWeekdayOpen(val value: String) : CafeInfoEditAction
    data class ChangeWeekdayClose(val value: String) : CafeInfoEditAction
    data class ChangeWeekendOpen(val value: String) : CafeInfoEditAction
    data class ChangeWeekendClose(val value: String) : CafeInfoEditAction
    data object ClickRepresentativeImage : CafeInfoEditAction
    data class SelectRepresentativeImage(val imageUrl: String) : CafeInfoEditAction
    data object ClickAddGalleryImage : CafeInfoEditAction
    data class AddGalleryImage(val imageUrl: String) : CafeInfoEditAction
    data object ClickPinLocation : CafeInfoEditAction
    data object ClickManageExceptionDates : CafeInfoEditAction
    data object DismissImageRequiredAlert : CafeInfoEditAction
    data object ClickSave : CafeInfoEditAction
    data object DismissInfoMessage : CafeInfoEditAction
}
